src/Features/Core/Portable/EmbeddedLanguages/AbstractEmbeddedLanguageDiagnosticAnalyzer.cs

```diff
@@ -1,70 +1,63 @@
 ﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
+using System;
 using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Linq;
+using Microsoft.CodeAnalysis.CodeStyle;
 using Microsoft.CodeAnalysis.Diagnostics;
-using Microsoft.CodeAnalysis.EmbeddedLanguages.LanguageServices;
 using Microsoft.CodeAnalysis.PooledObjects;
 
-namespace Microsoft.CodeAnalysis.EmbeddedLanguages
+namespace Microsoft.CodeAnalysis.Features.EmbeddedLanguages
 {
     internal abstract class AbstractEmbeddedLanguageDiagnosticAnalyzer : DiagnosticAnalyzer, IBuiltInAnalyzer
     {
-        private readonly ImmutableArray<IEmbeddedDiagnosticAnalyzer> _analyzers;
+        private readonly ImmutableArray<AbstractCodeStyleDiagnosticAnalyzer> _analyzers;
+        private readonly DiagnosticAnalyzerCategory _category;
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; }
 
         protected AbstractEmbeddedLanguageDiagnosticAnalyzer(
-            IEmbeddedLanguagesProvider languagesProvider)
+            IEmbeddedLanguageFeaturesProvider languagesProvider)
         {
             var supportedDiagnostics = ArrayBuilder<DiagnosticDescriptor>.GetInstance();
 
-            var analyzers = ArrayBuilder<IEmbeddedDiagnosticAnalyzer>.GetInstance();
+            var analyzers = ArrayBuilder<AbstractCodeStyleDiagnosticAnalyzer>.GetInstance();
 
-            if (languagesProvider != null)
+            var category = default(DiagnosticAnalyzerCategory?);
```

Comment:
>var category = default(DiagnosticAnalyzerCategory?); [](start = 12, length = 52)

nit: I found this unusual and it made me pause. I'd shift the type to the left-side and use `null` literal.

Reply:
IDE code (generally) uses 'var' everywhere possible. This just fits with that approach.  I'm ok with this as is.